main.go

```
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// Alert falco data structure
+type Alert struct {
+	Output       string    `json:"output"`
+	Priority     string    `json:"priority"`
+	Rule         string    `json:"rule"`
+	Time         time.Time `json:"time"`
+	OutputFields struct {
+		ContainerID              string      `json:"container.id"`
+		ContainerImageRepository interface{} `json:"container.image.repository"`
+		ContainerImageTag        interface{} `json:"container.image.tag"`
+		EvtTime                  int64       `json:"evt.time"`
+		FdName                   string      `json:"fd.name"`
+		K8SNsName                string      `json:"k8s.ns.name"`
+		K8SPodName               string      `json:"k8s.pod.name"`
+		ProcCmdline              string      `json:"proc.cmdline"`
+	} `json:"output_fields"`
+}
+
+func main() {
+	criticalNamespaces := map[string]bool{
+		"kube-system":     true,
+		"kube-public":     true,
+		"kube-node-lease": true,
+		"falco":           true,
+	}
```

I would actually fail early here if it was a critical namespace and skip passing it to the deletePod() function.

---

Can't both you and @bittrance  happy on this one :) 
He asked me earlier to break out to delete and if statement in a separate function. At least if I understood him correctly. 

The question is, should we test to see if we are in k8s first or should we check if we aren't in a protected namespace first?

---

I think a process should generally ascertain its ability to function properly before starting any work. Consider a scenario where it is difficult for me to provoke a Falco event that results in a dead pod. In that scenario, if we are taking decisions (i.e. early return on protected namespaces) before we done all the setup, it may take a long time before we discover that we have bad credentials.  This is basically the [fail-fast](https://www.agile-academy.com/en/agile-dictionary/fail-fast/) principle applied to software design.

---

This was what I meant: https://github.com/NissesSenap/falcosidekick-tekton/pull/1#discussion_r626053941

I was thinking that this could be done earlier, but making sure the credentials work seems like a good idea before. 👍